src/shared/sol-util-file.c

```diff
@@ -615,3 +616,81 @@ sol_util_file_get_basename(struct sol_str_slice path)
 
     return basename;
 }
+
+SOL_API ssize_t
+sol_util_write_file_slice(const char *path, struct sol_str_slice slice)
+{
+    FILE *fp;
+    size_t bytes;
+    int r;
+
+    SOL_NULL_CHECK(path, -EINVAL);
+
+    errno = 0;
```

Comment:
it's not necessary since if `fopen` fails it will return `NULL` and sets the `errno`